lib/runtime-c-api/src/lib.rs

```diff
@@ -0,0 +1,282 @@
+extern crate wasmer_runtime;
+extern crate wasmer_runtime_core;
+
+use libc::{c_char, c_int, int32_t, int64_t, uint32_t, uint8_t};
+use std::ffi::CStr;
+use std::slice;
+use std::str;
+use std::sync::Arc;
+use std::{ffi::c_void, mem, ptr};
+use wasmer_runtime::{ImportObject, Instance, Value};
+use wasmer_runtime_core::export::{Context, Export, FuncPointer};
+use wasmer_runtime_core::import::{LikeNamespace, Namespace};
+use wasmer_runtime_core::types::{FuncSig, Type};
+
+#[allow(non_camel_case_types)]
+pub struct wasmer_import_object_t();
+
+#[allow(non_camel_case_types)]
+pub struct wasmer_instance_t();
+
+#[allow(non_camel_case_types)]
+pub struct wasmer_instance_context_t();
+
+#[allow(non_camel_case_types)]
+#[no_mangle]
+#[repr(C)]
+pub enum wasmer_compile_result_t {
+    WASMER_COMPILE_OK = 1,
+    WASMER_COMPILE_ERROR = 2,
+}
+
+#[allow(non_camel_case_types)]
+#[no_mangle]
+#[repr(C)]
+pub enum wasmer_call_result_t {
+    WASMER_CALL_OK = 1,
+    WASMER_CALL_ERROR = 2,
+}
+
+#[repr(u32)]
+#[derive(Clone)]
+pub enum wasmer_value_tag {
+    WASM_I32,
+    WASM_I64,
+    WASM_F32,
+    WASM_F64,
+}
+
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub union wasmer_value {
+    I32: int32_t,
+    I64: int64_t,
+    F32: f32,
+    F64: f64,
+}
+
+#[repr(C)]
+#[derive(Clone)]
+pub struct wasmer_value_t {
+    tag: wasmer_value_tag,
+    value: wasmer_value,
+}
+
+#[repr(C)]
+#[derive(Clone)]
+pub struct wasmer_memory_t {
+    pub ptr: *mut uint8_t,
+    pub len: uint32_t,
+}
+
+#[no_mangle]
+pub extern "C" fn wasmer_import_object_new() -> *mut wasmer_import_object_t {
```

Comment:
It seems odd that this function returns the value, yet `wasmer_instantiate` takes it by reference?

Reply:
I think it’s because instantiate has a lot of scenarios with possible error results, but new import object doesn’t have any likely failure scenarios.